c/kastore.c

```diff
@@ -653,14 +655,18 @@ kastore_close(kastore_t *self)
             /* We only alloc memory for the keys and arrays in write mode */
             for (j = 0; j < self->num_items; j++) {
                 kas_safe_free(self->items[j].key);
-                kas_safe_free(self->items[j].array);
+                if (!self->items[j].ownership_taken) {
```

Comment:
Ooh, hold on - why are we checking this in write mode? This should only apply in read mode, so we can bug_assert on !ownership_taken.